macros/src/main/scala/io/github/hamsters/GenMacro.scala

```diff
@@ -0,0 +1,51 @@
+package io.github.hamsters
+
+import scala.collection.immutable.Seq
+import scala.meta._
+
+class GenMacro extends scala.annotation.StaticAnnotation {
```

Comment:
Can you make it clear in the macro name that it is a Hlist macro? Gen is too generic imho.

Reply:
Or maybe `@Derivation` as we may want to use this to derivate typeclasses from case classes in the future?

Reply:
I think @Derivation is too generic too, I prefer HList